flag.go

```diff
@@ -595,3 +596,27 @@ func withEnvHint(envVar, str string) string {
 	}
 	return str + envText
 }
+
+func stringifyFlag(f Flag) string {
+	fv := reflect.ValueOf(f)
+	placeholder, usage := unquoteUsage(fv.FieldByName("Usage").String())
```

Comment:
Hehe, I'm still curious what you think of using an interface here. Maybe creating a:

```go
type StringableFlag interface {
  NameString() string
  UsageString() string
  ValueString() string
  FormatValueHelp() function ...
}
```

Or something similar. Perhaps also adding additional methods to the interface to guard against other info the user may want to extract

I'm just wary of forcing the user to use reflection if they want to provide their own `FlagStringer` to do anything meaningful.